assimilation_code/modules/assimilation/assim_tools_mod.f90

```diff
@@ -469,10 +418,9 @@ subroutine filter_assim(ens_handle, obs_ens_handle, obs_seq, keys,           &
    endif
 endif
 
-!GSR open the dignostics file
-if(output_localization_diagnostics .and. my_task_id() == 0) then
+! Open the localization diagnostics file
+if(output_localization_diagnostics .and. my_task_id() == 0) &
   localization_unit = open_file(localization_diagnostics_file, action = 'append')
-endif
 
 ! For performance, make local copies of these settings which
 ! are really in the inflate derived type.
```

Comment:
I remember this and agree that it would be nice to remove. However, don't want to have to rerun all the timing tests, just in case, so will postpone for next refactor (coming sooner than you might think).